### PR TITLE
Implemented isinf to ivy and all backend

### LINKED ISSUE #146: isinf

### PR DESCRIPTION
**Pull request by taltrums**
test cases passing for tensorflow, numpy, torch, and jax
Fixes #146

### CODE REVIEW

**Review by saeedashrraf — COMMENTED** [review]
Hello @taltrums , thanks for your effort. This looks good. We will just wait for Dan's final confirmation before merging.

**Review by djl11 — APPROVED** [review]
lgtm!

**Pull request merged**
